onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/Shaders/bluestein_chirp.hlsl

```diff
@@ -0,0 +1,77 @@
+// TBUFFER is the data type to read from src and write to dst.
+// Arithmetic is always done in FP32.
+#if !defined(TBUFFER)
+#define TBUFFER float
+#endif
+
+RWStructuredBuffer<TBUFFER> chirp : register(u0);
+RWStructuredBuffer<TBUFFER> b : register(u1);
+
+cbuffer Constants
+{
+    uint StartIndex;
+    uint ElementCount;
+    uint DFTLength;
+    uint IsInverse;
+};
+
+uint NextPowerOf2(uint x) {
+  x--;
+  uint y = 1;
+  while (y <= x) {
+    y <<= 1;
+  }
+  return y;
+}
+
+
+// Returns the indices for the real and complex output uav
+uint2 ComputeDestIndex(uint n)
+{
+    return uint2(n*2, n*2+1);
+}
+
+float2 CalculateChirp(uint n, uint N, bool isInverse)
+{
+    static const float PI = 3.14159265f;
+    float direction = isInverse ? 1 : -1;
+    // chirp[n] = e^(i * direction * pi * n * n / N)
+    float theta = direction * PI * n * n / N;
+    return float2(cos(theta), sin(theta));
+}
+
+[numthreads(64, 1, 1)]
+void DFT(uint3 dtid : SV_DispatchThreadId)
+{
+    uint n = StartIndex + dtid.x;
```

Comment:
`elementIndex` instead of `n`?

Reply:
n is a common indexing variable in the DFT e^(-i * 2 * pi* n / N), or Chirp e^(-i * pi * n *n / N) equations.
elementIndex tends to make the maintainer think more in terms of array manipulation rather than the math of it.

Reply:
🤨 Is this a mathematicians vs software developers convention then? `i` is a common indexing variable in computer science, whereas `n` in computer science is almost always a count. So, `n` will probably be more confusing for future maintainers who speak computer science more than they speak DFT (because currently you're the only expert in DFT on our team :b).